src/shaka_scheme/system/core/vectors.cpp

```diff
@@ -0,0 +1,45 @@
+#include "shaka_scheme/system/core/vectors.hpp"
+
+namespace shaka {
+namespace core {
+
+/**
+ * @brief Converts a proper list into a vector.
+ * @param list A proper list, terminated with a null list
+ * @return A vector containing the same elements as the original list
+ */
+NodePtr list_to_vector(NodePtr list) {
+  // Find the length of the list.
+  std::size_t length = core::length(list);
+  shaka::Vector vector(length);

```

Comment:
Already using namespace `shaka` and namespace `core`

Reply:
`core::length` is unambiguous but also specific enough such that it's different from the slew of `length` functions out there. I can change this to `shaka::core::length` to make it more explicit.